recipes/llvm-core/all/conanfile.py

```diff
@@ -1,429 +1,491 @@
+        # https://releases.llvm.org/13.0.0/docs/CMake.html
+        cmake_variables = {
+            "LLVM_TARGETS_TO_BUILD": self._targets_to_build,
+            # See comment below on LLVM shared library builds
+            "LLVM_BUILD_LLVM_DYLIB": self.options.shared,
+            "LLVM_LINK_LLVM_DYLIB": self.options.shared,
+            "LLVM_DYLIB_COMPONENTS": self.options.components,
+            "LLVM_ABI_BREAKING_CHECKS": "WITH_ASSERTS",
+            "LLVM_INCLUDE_TOOLS": True,
+            "LLVM_INCLUDE_EXAMPLES": False,
+            "LLVM_INCLUDE_TESTS": False,
+            "LLVM_ENABLE_IDE": False,
+            "LLVM_ENABLE_EH": self.options.exceptions,
+            "LLVM_ENABLE_RTTI": self.options.rtti,
+            "LLVM_ENABLE_THREADS": self.options.threads,
+            "LLVM_ENABLE_LTO": self.options.lto,
+            "LLVM_STATIC_LINK_CXX_STDLIB": self.options.static_stdlib,
+            "LLVM_ENABLE_UNWIND_TABLES": self.options.unwind_tables,
+            "LLVM_ENABLE_EXPENSIVE_CHECKS": self.options.expensive_checks,
+            "LLVM_ENABLE_ASSERTIONS": self.settings.build_type,
+            "LLVM_USE_PERF": self.options.use_perf,
+            "LLVM_ENABLE_LIBEDIT": self.options.get_safe("with_libedit", False),
+            "LLVM_ENABLE_Z3_SOLVER": self.options.with_z3,
+            "LLVM_ENABLE_FFI": self.options.with_ffi,
+            "LLVM_ENABLE_ZLIB": "FORCE_ON" if self.options.with_zlib else False,
+            "LLVM_ENABLE_LIBXML2": "FORCE_ON" if self.options.with_xml2 else False,
+            "LLVM_ENABLE_TERMINFO": self.options.with_terminfo
+        }
 
-    def package_id(self):
-        del self.info.options.use_llvm_cmake_files
+        self._apply_resource_limits(cmake_variables)
 
-    def validate(self):
-        if self.options.shared:  # Shared builds disabled just due to the CI
-            message = 'Shared builds not currently supported'
-            raise ConanInvalidConfiguration(message)
-            # del self.options.fPIC
-        # if self.settings.os == 'Windows' and self.options.shared:
-        #     message = 'Shared builds not supported on Windows'
-        #     raise ConanInvalidConfiguration(message)
-        if self.options.exceptions and not self.options.rtti:
-            message = 'Cannot enable exceptions without rtti support'
-            raise ConanInvalidConfiguration(message)
-        self._supports_compiler()
-        if cross_building(self, skip_x64_x86=True):
-            raise ConanInvalidConfiguration('Cross-building not implemented')
+        # this capability is back-ported from LLVM 14.x
+        is_platform_ELF_based = self.settings.os in [
+            "Linux", "Android", "FreeBSD", "SunOS", "AIX", "Neutrino", "VxWorks"
+        ]
+        if is_platform_ELF_based:
+            self.output.info("ELF Platform Detected, optimizing memory usage during debug build linking.")
+            cmake_variables["LLVM_USE_SPLIT_DWARF"] = True
```

Comment:
This was necessary to get a debug build to succeed on the CCI build system. Without it I would have to deactivate all Debug builds. I believe the patch originated with one of the other LLVM recipes on which I based this one.

Reply:
Having said that - it's been a while since I tested a lot of these changes and there have been various changes to the recipe since then. It might take me some time to go back through the various configurations but I will check.

Reply:
I added it in my recipe: https://github.com/conan-io/conan-center-index/pull/17509/commits/e4bd6fa50f584775f6950e193cc2e08554e3e9dc

yes its not default but as far as I know was recommended in the llvm docs to reduce memory consumption during linking, at least for llvm 14+ https://releases.llvm.org/14.0.0/docs/CMake.html
> If enabled CMake will pass -gsplit-dwarf to the compiler. This option reduces link-time memory usage by reducing the amount of debug information that the linker needs to resolve. It is recommended for platforms using the ELF object format, like Linux systems when linker memory usage is too high.

Reply:
Thanks @planetmarshall  and @jusito for the added context.

If the big motivator is the Conan Center CI, and given that this affects Linux _debug_ builds - I personally think that we could probably avoid the complexity of backporting a feature from a newer version, and simply skipping that configuration in our CI (checking against the value of the environment variable, so that it is only disabled for this CI). 

Bear in mind that Linux debug binaries are generally less requested than release anyway.